providers/implementations/kdfs/hkdf.c

```diff
@@ -125,17 +132,28 @@ static void kdf_hkdf_free(void *vctx)
     KDF_HKDF *ctx = (KDF_HKDF *)vctx;
 
     if (ctx != NULL) {
-        kdf_hkdf_reset(ctx);
+        kdf_hkdf_reset_ex(vctx, 1);
         OPENSSL_free(ctx);
     }
 }
 
 static void kdf_hkdf_reset(void *vctx)
+{
+    kdf_hkdf_reset_ex(vctx, 0);
+}
```

Comment:
I think `kdf_hkdf_reset` is an internal function. If so, you can change it's declaration

Reply:
I can't because it's a provider callback function and must match the `OSSL_FUNC_kdf_reset_fn` signature.

Reply:
OK then. Thanks!